metadata_layer.py

```diff
@@ -574,56 +574,33 @@ def finalize(self, service, service_type, collection, start=NO_VALUE,
             self.collection_id = None
         else:
             self.collection_id = collection.id
-        if self.start is self.NO_VALUE:
+        if self.start is None:
             self.start = start
-        if self.finish is self.NO_VALUE:
-            if finish is self.NO_VALUE:
+        if self.finish is None:
+            if finish is None:
                 finish = datetime.datetime.utcnow()
```

Comment:
This bit is also duplicative of what happens in `Timestamp.stamp` and maybe we should take it out.